analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt

```diff
@@ -113,32 +112,58 @@ class NuGetSupport(
 
     private val registrationsBaseUrls: List<String> = getRegistrationsBaseUrls(definitionFile)
 
-    private val packageMap = mutableMapOf<Identifier, Pair<AllPackageData, Package>>()
-
     private inline fun <reified T> ObjectMapper.readValueFromUrl(url: String): T {
         val text = client.downloadText(url).getOrThrow()
         return readValue(text)
     }
 
-    private fun getAllPackageData(id: Identifier): AllPackageData {
-        // Note: The package name in the URL is case-sensitive and must be lower-case!
-        val lowerId = id.name.lowercase()
+    private fun getAllPackageData(
+        directDependencies: Collection<Identifier>
+    ): Pair<Map<Identifier, AllPackageData>, Set<OrtIssue>> {
+        val issues = mutableSetOf<OrtIssue>()
+        val result = mutableMapOf<Identifier, AllPackageData>()
+        val queue = LinkedList(directDependencies)
+
+        while (queue.isNotEmpty()) {
```

Comment:
@oheger-bosch can you do me a favor and review this code from a concurrency perspective?